hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetBootstrap.java

```diff
@@ -138,6 +150,51 @@ public static synchronized void executeJar(@Nonnull Supplier<JetInstance> suppli
         }
     }
 
+    private static void awaitJobsStarted() {
+        List<Job> submittedJobs = JetBootstrap.supplier.get().submittedJobs();
+        int submittedCount = submittedJobs.size();
+        if (submittedCount == 0) {
+            System.out.println("The JAR didn't submit any jobs.");
+            return;
+        }
+        int previousCount = -1;
+        while (true) {
+            uncheckRun(() -> Thread.sleep(JOB_START_CHECK_INTERVAL_MILLIS));
+            List<Job> startedJobs = submittedJobs.stream()
+                .filter(job -> !STARTUP_STATUSES.contains(job.getStatus()))
+                .collect(Collectors.toList());
+
+            submittedJobs = submittedJobs.stream()
+                    .filter(job -> !startedJobs.contains(job))
+                    .collect(Collectors.toList());
+
+            int remainingCount = submittedJobs.size();
+
+            if (submittedJobs.isEmpty() && remainingCount == previousCount) {
+                break;
+            }
+            if (remainingCount == previousCount) {
+                continue;
+            }
+            for (Job job : startedJobs) {
+                // The change of job statuses after the check above
+                // won't be a problem here. Because they cannot revert
+                // back to startup statuses.
+                if (job.getName() != null) {
+                    System.out.println("Job '" + job.getName() + "' is " + job.getStatus().name() + ".");
+                } else {
+                    System.out.println("Job " + job.getIdString() + " is " + job.getStatus().name() + ".");
+                }
```

Comment:
I think a date would be useful for debugging - e.g. when there is a large pause between submission and change to running state.